### DISCUSSION TITLE
Creating a Dynamo node

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#5419] [Ideas]
This is a follow-up to https://github.com/cyberbotics/webots/pull/5377
I advise to read the discussion we had there, to get an idea on reason between this new node.

The idea is to add a new node that will be able to : 
 - Output the speed of the joint, directly and/or by providing the output voltage.
 - Convert speed into electric power, using some chosen physical constant, (and maybe some sort of relation using table between input speed and output current)
 - By itself, apply a resistive force on the joint that is based by the usage of the Dynamo (the faster it turn, the more current it generate, the more resistive torque it generate)
 - Load a Battery

Usage : 
 - A wind mill, to study the impact of the resistive torque of the generator, in many wind condition
 - A light on a bike alimented by a dynamo (when driving, the light could light up, when stopping, the light shut of)
 - Current generation in a simulated gas engine, via the engine chain distribution
 - A self propelled robot, with a little wind mill on top !

I don't know if it would make sense to implement such a node, so I'm discussing that here.

**Comment by omichel** [comment]
To be frank, I do not see may use cases of this in robotics. So, this is not a priority for us unless we have a serious use case.

**Reply by ShuffleWire** [reply]
Yes, I understand that, maybe in the future !
